### PR TITLE
making stashcp a package not a module  (SOFTWARE-3799)

### PR DESCRIPTION
**Pull request by ddavila0**
When installed as a package, the option package_data can be used to put caches.json within the package directory. 
We still use data_files to put caches.json outside the package directory, in <sys.prefix>/share/stashcache, the later location of caches.json has precedence over the one inside the package directory and /etc/stashcache takes precedence over both of them.

/etc/stashcache/caches.json is not created by the installation process if needed has to be created by the user.

### CODE REVIEW

File: stashcp/__init__.py [hunk 1]
```diff
@@ -554,12 +554,12 @@ def get_best_stashcache():
...
-                       os.path.join(prefix, "etc/stashcache/caches.json"),
-                       os.path.join(prefix, "usr/share/stashcache/caches.json")]
-        if resource_string:
+        cache_files = [os.path.join(prefix, "etc/stashcache/caches.json"),
+                       os.path.join(prefix, "usr/share/stashcache/caches.json"),
+                       os.path.join(prefix, "usr/local/share/stashcache/caches.json")]
+        if resource_filename:
             try:
-                cache_files.insert(0, resource_string(__name__, 'caches.json'))
+		cache_files.append(resource_filename(__name__, 'caches.json'))
```
Comment (djw8605) [inline stashcp/__init__.py]: Why did this become un-indented?

**Comment by ddavila0** [comment]
What if we instead install the package (pip install StashCache/) and let it read the file from /share/stashcache ? I think that testing this way is more appropriate, what do you think?

**Comment by djw8605** [comment]
Agreed, install stashcache before testing it.

**Comment by djw8605** [comment]
Probably have to install `pip` first.

**Comment by ddavila0** [comment]
Tests are fine now, do everything else look fine?

**Pull request merged**